examples/with-playwright/app/index.html

```diff
@@ -8,9 +8,9 @@
       globalThis.process = { env: {} }
     </script>
     <script type="module" src="/browser.js"></script>
-    <script>
+    <script type="module">
       async function main() {
-        await navigator.serviceWorker.ready
```

Comment:
Do you know some technical details why this isn't the same thing? I was considering migrating to the `.ready` Promise internally in MSW to implement `worker.start()` Promise but it seems the two differ behavior-wise. Wonder if you know more.

Reply:
Maybe they are the same and the reason this PR works better is the inclusion of 
```
<script type="module">
```
If you take that out you run the script blocks in the wrong order, the one without type="module" then the service worker registration

I can change this PR up to only alter the script tag

Reply:
Another option is moving the main function to a separate file, like `main.js` and just adding `defer` to the script tag

```
<script type="module" src="/browser.js"></script>
<script defer src="./main.js"></script>
```

Reply:
hmm, nevermind, 2 separate issues.
They do seem different, but I'm not sure why

Reply:
My main concern is that these examples worked before. They also work now locally. They also work now in their own CI. The only place they fail is the MSW's `smoke-test` CI. By that judgment, I suspect the issue lies in MSW's job, not here. I may be wrong on this though. It started happening in 2.1.